spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -255,6 +257,42 @@ public static void applyQueryOptions(StructuredQuery.Builder builder, DatastoreQ
 						DatastoreTemplate.this.objectToKeyFactory)));
 	}
 
+	@Override
+	public <T> Map<String, T> findByIdAsMap(Key key, Class<T> valueType) {
+		Assert.notNull(key, "A non-null Key is required.");
+		Assert.notNull(valueType, "A non-null valueType is required.");
+
+		Entity entity = getDatastoreReadWriter().get(key);
+		if (entity == null) {
+			return null;
+		}
+		Map<String, T> map = new HashMap<>();
+		entity.getNames().forEach(
+				propertyName -> {
+					Value<?> value = entity.getValue(propertyName);
+					map.put(propertyName,
+							this.datastoreEntityConverter.getConversions().convertOnReadSingle(value.get(), valueType));
+				});
+		return map;
+	}
+
+	@Override
+	public <V> void writeMap(Key datastoreKey, Map<String, V> map) {
+		Assert.notNull(datastoreKey, "A non-null Key is required.");
+		Assert.notNull(map, "A non-null map is required.");
+
+		Builder builder = Entity.newBuilder(datastoreKey);
+		map.forEach(
```

Comment:
These two little anonymous functions calling `convertOnWriteSingle` are very simple, but would it be possible to use the existing `readAsMap` and `write` methods from `DefaultDatastoreEntityConverter` ?  

It seems like what we're doing here is just a small subset of what those bigger methods can handle, so maybe it would be possible to use those here?

Reply:
Makes sense for `readAsMap`, good catch! 
But it is not going to work with `write` because it uses  persistent properties.